numpy/lib/npyio.py

```diff
@@ -561,6 +561,20 @@ def savez(file, *args, **kwds):
     its list of arrays (with the ``.files`` attribute), and for the arrays
     themselves.
 
+    When saving dictionaries, the dictionary keys themselves are stored
+    as zip files under the zip directory. Therefore to avoid unintended
+    changes to the keys, one should avoid using key strings that are not
+    valid file names.
+
+    Examples include strings that begin with "/", or contain ".." or 
+    "./", such as '/test/tmp.txt', './test/tmp.txt', or 
+    'test/../tmp.txt'
+
+    Should keys that include such patterns be required, they must be 
+    properly encapsulated and or escaped using either double quotes or
+    backslashes where necessary. Note that backslashes will be 
+    doubled in the returned keys.
```

Comment:
This isn't good advice - this often won't be enough, especially on windows where backslash is a path separator. Realistically, we have two options here:

* Tell the user that they're not choosing string names, but (relative?) file names
* percent-encode the strings within `savez` and `loadz`, such that the user thinks they are working with raw strings. This might have compatibility ramifications